add custom pragma support for var and let symbols

jcosborn opened the pull request:
This passes all tests, but it changes the ast attached to var and let symbols, so might need further testing.  It also changes what getImpl returns for those symbols, so may break code using that.

cooldome commented:
Interesting functionality, but inconsistency it introduces to what is the Ast of the variable is a bit concerning.

I would rather say that getImpl should always return nkIdentDef  and Ast of variable symbol should also be nkIdentDef  in all cases. Yes, it is a breaking change that needs to be documented in the changelog, but it is still much better than writing if statements and helpers everywhere.

@Araq,
Any issue with I have proposed to @jcosborn?

Araq commented:
The upcoming 0.20 is about to finally spec the ``typed`` AST variants so I suppose it's fine to change it if there is finally a spec to adhere to.

krux02 commented:
I do like this improvement. can you give an example of an ast that will look differently after these changes and then mention it in the changelog?

jcosborn commented:
I've rebased on devel and updated the changelog.
For a declaration like
```nim
let a = 0
```
`getImpl` would currently return
```nim
IntLit 0
```
this patch would make it return
```nim
IdentDefs
  Sym "a"
  Empty
  IntLit 0
```

Araq commented:
Very nice but please add one more thing: An ``--oldast`` switch for backwards compat. We will use this switch for the other upcoming AST changes too then.

Araq commented on compiler/options.nim:
```diff
@@ -40,7 +40,8 @@ type                          # please make sure we have under 32 options
     optLaxStrings,
-    optNilSeqs
+    optNilSeqs,
+    optOldAst
```
Maybe this should be a ``TGlobalOption`` instead? Not sure though.

jcosborn replied:
I didn't really know either, I just copied what nilseqs did.  I suppose you could set it in some modules and not others, and it should still work ok.

Araq replied:
Ok.
